### ISSUE TITLE
[DataGrid] Rows "Jumps" when scrolling to bottom.

### ISSUE THREAD

**Issue by hinsxd** [mui-org/material-ui-x#414]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯
https://codesandbox.io/s/data-grid-bug-388qz
When scrolling to bottom, the rows "jumped" and unnecessary spacing appears at the bottom.

<img width="584" alt="bug" src="https://user-images.githubusercontent.com/3165635/95747720-2e39c680-0c99-11eb-81a7-50a59d2ee2d8.png">

## Expected Behavior 🤔
Scrolls normally to the bottom, with the last item sticking to the bottom.

## Steps to Reproduce 🕹
See codesandbox

Steps:

1.
2.
3.
4.

## Your Environment 🌎

| Tech          | Version |
| ------------- | ------- |
|@material-ui/data-grid"| "^4.0.0-alpha.6"|
| React         |    16.13.1     |
| Browser       |     macOS Chrome v85  |
| TypeScript    |         |
| etc.          |         |

**Comment by oliviertassinari** [comment]
> When scrolling to bottom, the rows "jumped" and unnecessary spacing appears at the bottom.

I can observe the spacing that shouldn't be present, however, what do you mean by "jump" do you have a recording?

**Comment by hinsxd** [comment]
Yes here it is. I sometimes encounter this but sometimes don't. 
https://imgur.com/Mqfpo0a

And this should be related too. I check DevTools and noticed that the last row is not rendered sometimes.
https://imgur.com/xXqpyVF

**Comment by hinsxd** [comment]
Is there a way to disable virtualization? Many users won't load more than 100 rows and virtualization seems overkill sometimes.

**Comment by oliviertassinari** [comment]
@hinsxd We could consider this option. However, I would recommend we wait for #193 to land before exploring it further.

**Comment by ZeeshanTamboli** [comment]
I will be looking into this. Let's see what I can find.

**Comment by royletron** [comment]
Hi @oliviertassinari I am also interested in disabling the virtualisation for small grids but with lots of UI elements. I am not entirely sure how it relates to #193 but would be eager to understand what is holding back the ability to set `rowBuffer` like we currently set `columnBuffer`?

**Comment by oliviertassinari** [comment]
>  if we have a monitor high enough to show all 100 rows

@royletron this sounds like #270. I was advocating that the overscan logic is suboptimal.
Do you have a codesandbox with the display conditions so we can investigate?

**Comment by royletron** [comment]
@oliviertassinari I'll have a go and get back to you. We are using a few custom renderers on our columns which allow for interaction so it seems mounting those from the buffer is expensive.

**Comment by royletron** [comment]
@oliviertassinari check out https://codesandbox.io/s/mui-x-414-xlhsq 

Particularly scrolling to the bottom of the grid.
Like I say, I understand that this is quite intensive, but our solution is to limit the number of rows shown at once, so controlling the virtualisation would be ideal.

**Comment by oliviertassinari** [comment]
@royletron Perfect. Note that `renderCell` expects a render function, not a component. The hooks work by chance inside them.

We have little room to improve. The core components weren't designed to render smoothly with x100 instances x10 Hz. If it was the objective, we would have made significantly different choices. I would recommend:

- Material-UI: That we reduce the rendering window #270
- Developers: Rendering simpler components, not an edit all move.
- Material-UI/Developers: Testing different virtualization technics but I don't think that any will work with the currently rendered components
 
As for making the core component render more smoothly. You are using v4, they are getting a bit slower in v5 because we move from JSS to emotion/styled-components. Nothing major when you have a few instances in a pages, but definitely noticeable when you render x100 multiple time per second.

**Comment by oliviertassinari** [comment]
> and unnecessary spacing appears at the bottom.

I believe we didn't solve this part of the issue. You can see it it on https://master--material-ui-x.netlify.app/components/data-grid/demo/.
